### PR TITLE
Additional optimizations for the fallout_GPU subroutine.

### PR DESCRIPTION
**Pull request by sherimickelson**

### CODE REVIEW

**Comment by sherimickelson** [comment]
[Optimize fallout_GPU on GPU](https://trello.com/c/eaxe6CUe/26-optimize-falloutgpu-on-gpu)

File: src/kessler.F [hunk 1]
```diff
@@ -736,80 +736,101 @@ subroutine fallout_GPU(dt,train,ruh,rvh,zh,mh,mf,rain,prate,rr,rho,   &
...
+    !$acc loop gang vector collapse(3) 
+    do k=1,nk
+    do j=1,nj
     do i=1,ni
-      do n=1,nfall(i)
-        do k=1,nk
-         rq(k)=rho(i,j,k)*vq(i,j,k)*max(0.0,q3d(i,j,k))
-          rq(nk+1)=0.0 ! Added here from below by SM
+    ll = nfall(i,j)
+    do n=1,ll
```
Comment (johnmauff) [inline src/kessler.F]: Why does the i and n loop exist in this code.  It appears to be duplicating assignments

Comment (sherimickelson) [reply src/kessler.F]: I removed the loop and I verified that the answers did not change with that modification.

File: src/kessler.F [hunk 2]
```diff
@@ -736,80 +736,98 @@ subroutine fallout_GPU(dt,train,ruh,rvh,zh,mh,mf,rain,prate,rr,rho,   &
...
     enddo
-    !$acc loop collapse(3) gang vector
+    enddo
+    !$acc end parallel
+
+    !$acc parallel default(present) private(i,j,k,n)
+    !$acc loop gang vector collapse(3) 
+    do k=1,nk
+    do j=1,nj
     do i=1,ni
```
Comment (johnmauff) [inline src/kessler.F:797]: What is the i-loop doing  here?  Is this a bug that needs to be discussed with George?

Comment (sherimickelson) [reply src/kessler.F:797]: I'm not sure if I understand your question.  Do you have time to talk about this?

Comment (sherimickelson) [reply src/kessler.F:797]: The latest commit should address this issue.

**Review by johnmauff — APPROVED** [review]

**Comment by johnmauff** [comment]
Thanks for addressing my concerns

**Pull request merged**